pkg/networkaware/networkoverhead/networkoverhead.go

```diff
@@ -0,0 +1,577 @@
+
+	// Create map for cost / destinations. Search for requirements faster...
+	var costMap = make(map[networkawareutil.CostKey]int64)
+
+	// Populate costMap
+	populateCostMap(no, costMap, networkTopology, region, zone)
+
+	klog.V(6).Info("costMap: ", costMap)
+
+	var cost int64 = 0
+	var ok error
+
+	// Get accumulated cost based on pod dependencies
+	cost, ok = getAccumulatedCost(scheduledList, dependencyList, nodeName, region, zone, costMap, no)
+	if ok != nil {
+		return score, framework.NewStatus(framework.Error, fmt.Sprintf("getting pod hostname from Snapshot: %v", ok))
+	}
+
+	// Return Accumulated Cost as score
+	score = cost
+
+	klog.V(4).Infof("pod:%s; node:%s; finalScore=%d", pod.GetName(), nodeName, score)
+	return score, framework.NewStatus(framework.Success, "Accumulated cost added as score, normalization ensures lower costs are favored")
+}
+
+// NormalizeScore : normalize scores since lower scores correspond to lower latency
+func (no *NetworkOverhead) NormalizeScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
+	klog.V(4).Infof("Scores before normalization: %s", scores)
+
+	// Get Min and Max Scores to normalize between framework.MaxNodeScore and framework.MinNodeScore
+	minCost, maxCost := getMinMaxScores(scores)
+
+	// If all nodes were given the minimum score, return
+	if minCost == 0 && maxCost == 0 {
+		return nil
+	}
+
+	var normCost float64
+	for i := range scores {
+		if maxCost != minCost { // If max != min
+			// node_normalized_cost = MAX_SCORE * ( ( nodeScore - minCost) / (maxCost - minCost)
+			// nodeScore = MAX_SCORE - node_normalized_cost
+			normCost = float64(framework.MaxNodeScore) * float64(scores[i].Score-minCost) / float64(maxCost-minCost)
+			scores[i].Score = framework.MaxNodeScore - int64(normCost)
+		} else { // If maxCost = minCost, avoid division by 0
+			normCost = float64(scores[i].Score - minCost)
+			scores[i].Score = framework.MaxNodeScore - int64(normCost)
+		}
+	}
+	klog.V(4).Infof("Scores after normalization: %s", scores)
+	return nil
+}
+
+// MinMax : get min and max scores from NodeScoreList
+func getMinMaxScores(scores framework.NodeScoreList) (int64, int64) {
+	var max int64 = math.MinInt64 // Set to min value
+	var min int64 = math.MaxInt64 // Set to max value
+
+	for _, nodeScore := range scores {
+		if nodeScore.Score > max {
+			max = nodeScore.Score
+		}
+		if nodeScore.Score < min {
+			min = nodeScore.Score
+		}
+	}
+
+	return min, max
+}
+
+// sortNetworkTopologyCosts : sort costs if manual weights were selected
+func sortNetworkTopologyCosts(no *NetworkOverhead, networkTopology *ntv1alpha1.NetworkTopology) {
+	if no.weightsName != ntv1alpha1.NetworkTopologyNetperfCosts { // Manual weights were selected
+		for _, w := range networkTopology.Spec.Weights {
+			// Sort Costs by TopologyKey, might not be sorted since were manually defined
+			sort.Sort(networkawareutil.ByTopologyKey(w.TopologyList))
+		}
+	}
+	return
```

Comment:
this is redundant.

Reply:
What do you mean exactly here? Essentially, if costs are manually added, we check if they are alphabetically sorted since we apply binary search to get costs from a specific topology key.

Reply:
I mean the line `return` can be removed as this function doesn't need to return anything.